Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py

```diff
@@ -800,6 +801,71 @@ def update_remote_system_command(client: Client, args: Dict[str, Any], mirror_ta
     return new_incident_id
 
 
+def get_and_dedup_incidents(client: Client, incidents_last_fetch_ids: list[dict],
+                            query: str, max_results: int, last_fetch: Union[str, int]) -> tuple[list[dict], list[dict]]:
+    """ get incidents and dedup  the incidents response.
+
+    Cases:
+    1.  Empty incidents list (no new incidents received from API response).
+        Meaning: Usually means there are not any more incidents to fetch at the moment.
+        Handle: Return empty list of incidents and the unchanged list of 'incidents_last_fetch_ids' for next run.
+
+    2.  The response include incidents from the previous fetch cycle.
+        Meaning: There are potentially more incidents with the same timestamp.
+        Handle: Get more incidents util the number of the incident equal to the requested max fetch_limit.
+        Add the list of fetched incidents IDs to current 'incidents_last_fetch_ids' from last run,
+        return list of new incidents and updated list of 'incidents_last_fetch_ids' for next run.
+
+    3.  Most recent incident has later timestamp then other incidents in the response.
+        Meaning: This is the normal case where incidents in the response have different timestamps.
+        Handle: Return list of new incidents and a list of 'new_ids' containing only IDs of
+                incidents with identical latest time for next run.
+
+    Args:
+        incidents (list[dict]): List of incidents from the current fetch response.
+        incidents_last_fetch_ids (list[dict]): List of IDs of incidents from last fetch cycle.
+
+    Returns:
+        tuple[list[dict], list[str]: The list of dedup incidents and ID list of incidents of current fetch.
+    """
+    last_fatched_incident = (dateparser.parse(incidents_last_fetch_ids[-1]["created"])
+                             if incidents_last_fetch_ids else None)
+    incidents = client.search_incidents(
+        query=query,
+        max_results=max_results,
+        start_time=last_fetch,
+        field="created",
+    )
+    new_incidents: list = []
+    page = 1
+    # Case 1: Empty response len(incidents) == 0
+    while len(incidents) > 0 and len(new_incidents) < max_results:
```

Comment:
Why not use *for* loop instead and remove the first case?
You should change these switch cases' structure to be clearer and less code.